src/module/rules/rule-element/base-speed.ts

```diff
@@ -40,12 +33,22 @@ class BaseSpeedRuleElement extends RuleElementPF2e {
 
     override beforePrepareData(): void {
         if (this.ignored) return;
-        const speed = this.#createMovementType();
-        const synthetics = (this.actor.synthetics.movementTypes[this.selector] ??= []);
+        
+        const resolved = this.resolveInjectedProperties(this.selector);
+
+        let speedType: MovementType = "land"
+        if (!tupleHasValue(MOVEMENT_TYPES, resolved)) {
+            this.failValidation("Unrecognized or missing selector");
```

Comment:
You need to return here, or validation-failing data will get added to the synthetics object.

Reply:
I figured we'd just add "land", same as original code. In any event, fixed.